google/cloud/aiplatform/utils/featurestore_utils.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+from typing import Optional, Tuple
+
+from google.cloud.aiplatform.compat.services import featurestore_service_client
+
+CompatFeaturestoreServiceClient = featurestore_service_client.FeaturestoreServiceClient
+
+RESOURCE_ID_PATTERN_REGEX = r"[a-z_][a-z0-9_]{0,59}"
+
+
+def validate_id(resource_id: str) -> bool:
+    """Validates feature store resource ID pattern."""
+    return bool(re.compile(r"^" + RESOURCE_ID_PATTERN_REGEX + r"$").match(resource_id))
+
+
+def validate_and_get_featurestore_resource_id(featurestore_name: str) -> str:
```

Comment:
This seems to only be used in the `FeatureStore` constructor `EntityType.list`. Both also rely on `utils.full_resource_name` which does similar validation. What is the benefit of using this approach? Is OK to just rely on `utils.full_resource_name`.